### DISCUSSION TITLE
Per-chapter bibliography for PDF book

### DISCUSSION THREAD

**Discussion by bbartholdy** [quarto-dev/quarto-cli#712] [Q&A] [answered]
Hi all,

I would like to be able to have my references at the end of each chapter, and only the references that were used in the chapter.

It's possible to do this for the HTML output with the `reference-location: section` yaml option, but this doesn't seem to carry over to the PDF output. I have also tried using `::: {#refs} :::` at the end of each chapter, but that prints the whole bibliography for the entire book.

Is there an easy option to do this that I missed in the documentation? If not, I can also submit an issue for a feature request.

**Comment by jjallaire** [comment, marked as answer]
It look like there is a path to do this this filter: https://github.com/pandoc/lua-filters/tree/master/section-refs

However, I don't know if this works straight away with Quarto. @cderv How easy would it be to make this work w/ Quarto and do you think we should just build it in so that it can be activated with a simple option?

**Reply by cderv** [reply]
I'll look into this to make it work. 

If this is possible to do that for HTML with `reference-location: section`, it would probably be interesting to allow this for other formats thanks to the filter - that is one of the advantage of Quarto being able to bundle Lua filter to improve Pandoc behavior IMO.

**Reply by bbartholdy** [reply]
The section-refs lua filter does kinda work, but it's still printing the bibliography at the end of the book in addition to at the end of each chapter, and I don't see a way to just suppress the bibliography at the end of the book. The `suppress-bibliography: true` option also removes the chapter references.

**Reply by cderv** [reply]
Thanks for additional information. This means we may need to adapt considering some of our internal processing. I'll have a look

**Reply by bbartholdy** [reply]
I re-read the lua filter README and I needed to change the `bibliography` yaml option to `section-refs-bibliography` (oops...). This removes the bibliography section at the back of the PDF output. HOWEVER, it also breaks the references in the HTML output, so I had to add the `bibliography` option under the HTML output. The following gives the desired behaviour in both outputs:

```yaml
format:
  html:
    bibliography: [book.bib]
  
  pdf:
    documentclass: scrbook
    pdf-engine: xelatex
    cite-method: citeproc
    section-refs-bibliography: book.bib
    filters: 
      - section-refs.lua # needs pandoc version >= 2.11 to work 
```

I hope this helps. It would be awesome to have a 'switch' to turn on per-chapter references in both html and pdf outputs! :smiley:

**Comment by jjallaire** [comment]
@dragonstyle It seems like this filter does indeed do most of what's necessary. I'd second the idea of formally incorporating this into Quarto.
